### PR TITLE
[VMware] Improve volume file search on the datastore while computing the VM snapshot chain size.

### PR DESCRIPTION
**Pull request by sureshanaparti**
### Description

This PR improves the volume file search on the datastore while computing the VM snapshot chain size in VMware. The file search is now performed in the VM directory on the datastore, instead of all the directories on the datastore (which is leading to incorrect VM snapshot chain size computation when VM id has 6 digits).

<!--- Describe your changes in DETAIL - And how has behaviour functionally changed. -->

<!-- For new features, provide link to FS, dev ML discussion etc. -->
<!-- In case of bug fix, the expected and actual behaviours, steps to reproduce. -->

<!-- When "Fixes: #<id>" is specified, the issue/PR will automatically be closed when this PR gets merged -->
<!-- For addressing multiple issues/PRs, use multiple "Fixes: #<id>" -->
<!-- Fixes: # -->

<!--- ********************************************************************************* -->
<!--- NOTE: AUTOMATATION USES THE DESCRIPTIONS TO SET LABELS AND PRODUCE DOCUMENTATION. -->
<!--- PLEASE PUT AN 'X' in only **ONE** box -->
<!--- ********************************************************************************* -->

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

### Feature/Enhancement Scale or Bug Severity

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial


### Screenshots (if appropriate):


### How Has This Been Tested?
<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

Manually tested create and delete VM snapshot operations in VMware, with 6 digit VM ids (set the VM instance sequence manually using the sql: "UPDATE cloud.sequence SET value = 100000 WHERE name = ‘vm_instance_seq’").

- After Create Snapshot:

```
MariaDB [cloud]> SELECT vol.id as volume_id, vol.uuid as volume_uuid, vol.name, vol.volume_type, vol.state, vol.path, vol.size, vol.vm_snapshot_chain_size, vol.chain_info, vol.instance_id, vm.instance_name as instance_name, vm.name as vm_name FROM volumes vol, vm_instance vm WHERE vol.instance_id = vm.id AND vm.id >= 100000;
+-----------+--------------------------------------+-------------+-------------+-------+----------------------------------+------------+------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------+---------------+---------+
| volume_id | volume_uuid                          | name        | volume_type | state | path                             | size       | vm_snapshot_chain_size | chain_info                                                                                                                                                                                                                   | instance_id | instance_name | vm_name |
+-----------+--------------------------------------+-------------+-------------+-------+----------------------------------+------------+------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------+---------------+---------+

|         6 | 61c4cfa4-33b1-4787-a48f-12167433a404 | ROOT-100001 | ROOT        | Ready | ROOT-000001                             | 2147483648 |              167801095 | {"diskDeviceBusName":"ide0:1","diskChain":["[50da2bc6a157338088dbbf52748b14cf] i-2-100001-VM/ROOT-100001.vmdk","[50da2bc6a157338088dbbf52748b14cf] 16dbb68180af34b5a9589699216e6c75/16dbb68180af34b5a9589699216e6c75.vmdk"]} |      100001 | i-2-100001-VM | vm03    |
|         7 | 483c4533-f8fd-45f2-8b8c-6d3aa4338564 | DATA-100001 | DATADISK    | Ready | 1b0ea534ebcf4ea6b22053ae608c876a-000001 | 3221225472 |             3221225472 | {"diskDeviceBusName":"scsi0:0","diskChain":["[50da2bc6a157338088dbbf52748b14cf] i-2-100001-VM/1b0ea534ebcf4ea6b22053ae608c876a.vmdk"]}                                                                                       |      100001 | i-2-100001-VM | vm03    |


+-----------+--------------------------------------+-------------+-------------+-------+----------------------------------+------------+------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------+---------------+---------+

```

- After Delete Snapshot:

```
MariaDB [cloud]> SELECT vol.id as volume_id, vol.uuid as volume_uuid, vol.name, vol.volume_type, vol.state, vol.path, vol.size, vol.vm_snapshot_chain_size, vol.chain_info, vol.instance_id, vm.instance_name as instance_name, vm.name as vm_name FROM volumes vol, vm_instance vm WHERE vol.instance_id = vm.id AND vm.id >= 100000;
+-----------+--------------------------------------+-------------+-------------+-------+----------------------------------+------------+------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------+---------------+---------+
| volume_id | volume_uuid                          | name        | volume_type | state | path                             | size       | vm_snapshot_chain_size | chain_info                                                                                                                                                                                                                   | instance_id | instance_name | vm_name |
+-----------+--------------------------------------+-------------+-------------+-------+----------------------------------+------------+------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------+---------------+---------+

|         6 | 61c4cfa4-33b1-4787-a48f-12167433a404 | ROOT-100001 | ROOT        | Ready | ROOT-100001                      | 2147483648 |                      0 | {"diskDeviceBusName":"ide0:1","diskChain":["[50da2bc6a157338088dbbf52748b14cf] i-2-100001-VM/ROOT-100001.vmdk","[50da2bc6a157338088dbbf52748b14cf] 16dbb68180af34b5a9589699216e6c75/16dbb68180af34b5a9589699216e6c75.vmdk"]} |      100001 | i-2-100001-VM | vm03    |
|         7 | 483c4533-f8fd-45f2-8b8c-6d3aa4338564 | DATA-100001 | DATADISK    | Ready | 1b0ea534ebcf4ea6b22053ae608c876a | 3221225472 |                      0 | {"diskDeviceBusName":"scsi0:0","diskChain":["[50da2bc6a157338088dbbf52748b14cf] i-2-100001-VM/1b0ea534ebcf4ea6b22053ae608c876a.vmdk"]}                                                                                       |      100001 | i-2-100001-VM | vm03    |


+-----------+--------------------------------------+-------------+-------------+-------+----------------------------------+------------+------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------+---------------+---------+

```


<!-- Please read the [CONTRIBUTING](https://github.com/apache/cloudstack/blob/main/CONTRIBUTING.md) document -->

### CODE REVIEW

**Comment by sureshanaparti** [comment]
@blueorangutan package

File: plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/manager/VmwareStorageManagerImpl.java [hunk 1]
```diff
@@ -1306,12 +1307,12 @@ private void setVolumeToPathAndSize(List<VolumeObjectTO> volumeTOs, Map<String,
             // get volume's chain size for this VM snapshot; exclude current volume vdisk
             DataStoreTO store = volumeTO.getDataStore();
             ManagedObjectReference morDs = getDatastoreAsManagedObjectReference(baseName, hyperHost, store);
-            long size = getVMSnapshotChainSize(context, hyperHost, baseName + ".vmdk", morDs, newPath);
-            size = getVMSnapshotChainSize(context, hyperHost, baseName + "-*.vmdk", morDs, newPath);
+            long size = getVMSnapshotChainSize(context, hyperHost, baseName + ".vmdk", morDs, newPath, vmName);
```
Comment (Pearl1594) [inline plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/manager/VmwareStorageManagerImpl.java:1310]: shouldn't this call be removed @sureshanaparti?

Comment (sureshanaparti) [reply plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/manager/VmwareStorageManagerImpl.java:1310]: yes @Pearl1594, already created a separate PR (https://github.com/apache/cloudstack/pull/5720).

**Comment by sureshanaparti** [comment]
@blueorangutan test centos7 vmware-67u3

**Review by nvazquez — APPROVED** [review]
Code LGTM, have not tested it

**Comment by sureshanaparti** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test centos7 vmware-70u1

**Review by Pearl1594 — APPROVED** [review]
Verified fix - file search restricted to the respective VM's folder. LGTM.

**Comment by andrijapanicsb** [comment]
@sureshanaparti does this handle detached volumes well? Afaik, the detached volumes are moved to a different folder than the folder where the VM is located (but, I might be wrong if this has changed recently).

**Pull request merged**